server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -235,13 +237,20 @@ public RemoteClusterStateManifestInfo writeFullMetadata(ClusterState clusterStat
             isPublicationEnabled,
             isPublicationEnabled ? clusterState.customs() : Collections.emptyMap(),
             isPublicationEnabled,
-            remoteRoutingTableService.getIndicesRouting(clusterState.getRoutingTable())
+            remoteRoutingTableService.getIndicesRouting(clusterState.getRoutingTable()),
+            null
         );
+        ClusterStateDiffManifest clusterStateDiffManifest = new ClusterStateDiffManifest(clusterState, ClusterState.EMPTY_STATE, null);
+        if (uploadedMetadataResults.uploadedIndicesRoutingDiffMetadata != null) {
+            clusterStateDiffManifest.setIndicesRoutingDiffPath(
+                uploadedMetadataResults.uploadedIndicesRoutingDiffMetadata.getUploadedFilename()
+            );
+        }
```

Comment:
Should we get the filename first and then create diff manifest, so we can avoid creating with null value and then going with setter.